### PR TITLE
COG-296-CANCEL button in User Profile page does not appear to cancel the changes

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**

### CODE REVIEW

File: app/javascript/views/userDetail/UserDetail.jsx [hunk 1]
```diff
@@ -52,9 +52,12 @@ export default class UserDetail extends Component {
   };
 
   onEditClick = () => {
-    this.state.isEdit === false
-      ? this.setState({ isEdit: true })
-      : this.setState({ isEdit: false });
+    this.setState({ isEdit: true });
```
Comment (kelfink) [inline app/javascript/views/userDetail/UserDetail.jsx]: It was toggling, before (second click sets edit to false....). But I think the button gets disabled when we get into edit mode.  So this is good simplification, yes?

File: app/javascript/views/userDetail/UserDetail.jsx [hunk 2]
```diff
@@ -52,9 +52,12 @@ export default class UserDetail extends Component {
...
+  };
+
+  onCancel = () => {
+    this.setState({ isEdit: false });
+    this.props.actions.fetchDetailsActions();
```
Comment (kelfink) [inline app/javascript/views/userDetail/UserDetail.jsx]: Yes, go reload the record.  This actually answers my previous question. We need to have different actions on "cancel". which reloads, versus clicking edit, which does not.

File: app/javascript/views/userDetail/UserDetail.jsx [hunk 3]
```diff
@@ -78,7 +81,7 @@ export default class UserDetail extends Component {
                 selectedPermissions={this.formattedPermissions(
                   this.state.details.permissions
                 )}
-                onCancel={this.onEditClick}
+                onCancel={this.onCancel}
```
Comment (kelfink) [inline app/javascript/views/userDetail/UserDetail.jsx]: Right, and completing the shift, the cancel button now performs the reload and switch back to view mode.  good.

File: app/javascript/views/userDetail/UserDetail.test.js [hunk 4]
```diff
@@ -46,8 +46,6 @@ describe('UserDetail', () => {
         const instance = wrapper.instance();
         instance.onEditClick();
         expect(instance.state.isEdit).toEqual(true);
```
Comment (kelfink) [inline app/javascript/views/userDetail/UserDetail.test.js]: SO we don't expect it to toggle try/false anymore, but we lack a test of the new method onCancel

Comment (Sushmareddy-OSI) [reply app/javascript/views/userDetail/UserDetail.test.js]: I tried testing onCancel but it throwing another error saying that action is not a function.

File: app/javascript/views/userDetail/UserDetail.jsx (hunk 2 above)
Comment (kelfink) [reply app/javascript/views/userDetail/UserDetail.jsx]: I don't see a unit test for onCancel.  Also we have an acceptance test which can now (after clicking cancel). show that the record gets restored.  That acceptance test can be a backlog item, but needs to be written up.  I'll do that.

**Comment by kelfink** [comment]
We are skipping the unit test requested but will circle back to it with a technical debt.

**Pull request merged**